File: src/org/usfirst/frc5293/translations/camera/CameraEngine.java
package org.usfirst.frc5293.translations.camera;

import org.usfirst.frc5293.Input;
import org.usfirst.frc5293.input.Camera;
import org.usfirst.frc5293.translations.util.Point;
import org.usfirst.frc5293.translations.util.StreamingTranslationEngine;

public class CameraEngine extends StreamingTranslationEngine<Point> {

    private static CameraEngine instance;

    public static CameraEngine getInstance() {
        if (instance == null) {
            instance = new CameraEngine();
        }
        return instance;
    }

    //

    private final Camera input;

    private CameraEngine() {
        super(CameraPipeline.getInstance());
        input = Input.getCamera();
    }

    @Override
    protected Point getInitial() {
        Point state = new Point();

        state.x = input.getJoystick().getTwist();
        state.y = input.getJoystick().getY();

        return state;
    }

}

File: src/org/usfirst/frc5293/translations/camera/CameraEngine.kt
package org.usfirst.frc5293.translations.camera

import org.usfirst.frc5293.Input
import org.usfirst.frc5293.translations.util.Point
import org.usfirst.frc5293.translations.util.StreamingTranslationEngine
import org.usfirst.frc5293.util.Initializable
import org.usfirst.frc5293.util.LazyByRequest
import org.usfirst.frc5293.util.LazySource
import org.usfirst.frc5293.util.lazyByRequest

object CameraEngine : StreamingTranslationEngine<Point>(CameraPipeline) {

    override val initial =
            Point(x = Input.camera.joystick.twist,
                  y = Input.camera.joystick.y)
}
